src/core/settings/qgssettingsregistry.h

```diff
@@ -0,0 +1,83 @@
+/***************************************************************************
+  qgssettingsregistry.h
+  --------------------------------------
+  Date                 : February 2021
+  Copyright            : (C) 2021 by Damiano Lombardi
+  Email                : damiano at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSSETTINGSREGISTRY_H
+#define QGSSETTINGSREGISTRY_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgssettingsentry.h"
+
+#include <QMap>
+
+/**
+ * \ingroup core
+ * \class QgsSettingsRegistry
+ * QgsSettingsRegistry is used for settings introspection and collects a
+ * list of child QgsSettingsRegistry and a list of child QgsSettingsRegistry
+ *
+ * \since QGIS 3.20
+ */
+class CORE_EXPORT QgsSettingsRegistry
+{
+  public:
+
+    /**
+     * Constructor for QgsSettingsRegistry.
+     */
+    QgsSettingsRegistry();
+
+    /**
+     * Destructor for QgsSettingsRegistry.
+     */
+    virtual ~QgsSettingsRegistry();
+
+    /**
+     * Add \a settingsEntry to the register.
+     */
+    void addSettingsEntry( const QgsSettingsEntryBase *settingsEntry );
```

Comment:
should this be protected?

Reply:
If we make it protected the writers of plugin would be forced to subclass QgsSettingsRegister. Shouldn't we let them the choice between subclassing or using it like in the example of the first comment?
Or are you thinking about making impossible to plugin authors to "inject" settings in other registry then its own? Then I would agree it should be protected.

Reply:
yes this is my concern. This would be very temptable to add settings to core from a plugin.

Reply:
I have made it protected but it does not make any differece, because I guess protected functions in cpp becomes public in python?

Reply:
no, it has been made protected: https://github.com/qgis/QGIS/pull/42860/files#diff-b4827b0b3c7c32b1f618b43065bdbedcdc0037568f6435143f35fc093a10f96dR58
or am I missing something?